File: src/test/kotlin/domain/TextUtilsPostProcessTest.java
package domain;

import org.junit.Assert;
import org.junit.Test;

import java.util.Arrays;
import java.util.List;

/**
 * Created by Zahar on 16.05.16.
 */
public class TextUtilsPostProcessTest {

	@Test
	public void testComplex() {
		String rawText = "qwer asdf qwer or qwer zxc";
		List<String> dictionary = Arrays.asList("qwer or qwer", "qwer asdf");

		String strings = TextUtils.Companion.preProcessText(dictionary, rawText);
		Assert.assertEquals("|qwer asdf| |qwer or qwer| zxc", strings);
	}
}

File: src/test/kotlin/domain/TextUtilsPostProcessTest.kt
package domain

import org.assertj.core.api.Assertions
import org.junit.Test

import java.util.Arrays

/**
 * Created by Zahar on 16.05.16.
 */
class TextUtilsPostProcessTest {

	@Test
	fun testComplex() {
		val words = Arrays.asList("|Magic", "The", "Gathering|", "|Hearthstone", "Heroes", "of", "Warcraft|")

		val wordsqwer: List<String> = TextUtils.postProcessText(words)
		Assertions.assertThat(wordsqwer).contains("Magic The Gathering", "Hearthstone Heroes of Warcraft").hasSize(2)
	}

	@Test
	fun testComplex2() {
		val words = Arrays.asList("qwer", "asdf", "|qwer", "or", "qwer|", "zxc")

		val wordsqwer = TextUtils.postProcessText(words)
		Assertions.assertThat(wordsqwer).contains("qwer", "asdf", "qwer or qwer", "zxc").hasSize(4)
	}
}
